dvc/remote/local/__init__.py

```diff
@@ -352,12 +359,21 @@ def status(self, checksum_infos, remote, jobs=None, show_checksums=False):
 
         progress.update_target(title, 30, 100)
 
+        local_exists = self.cache_exists(md5s)
+
+        # This is a performance optimization. We can safely assume that,
+        # if the resources that we want to fetch are already cached,
+        # there's no need to check the remote storage for the existance of
+        # those files.
+        if download and local_exists == md5s:
+            self._fill_statuses(ret, local_exists, local_exists)
```

Comment:
Also, no need to copy-paste code. You have this down below already, just use `if` for `remote_exists`.

Reply:
@efiop , assigning `remote_exists` is triggering `remote.cache_exists(md5s)` and this is what we want to avoid, right?

In my patch I'm testing that if we already have the files that we want to check `local_exists == md5s` then return the status object `self._fill_statuses(ret, local_exists, local_exists)`.

Reply:
Yes, I'm just saying that the code could be optimized to not copypaste stuff. How about something like:
```
        progress.update_target(title, 30, 100)               
                                                             
        local_exists = self.cache_exists(md5s)               
                                                             
        progress.update_target(title, 40, 100)               
                                                             
        if download and local_exists == md5s:                
            remote_exists = local_exists                     
        else:                                                
            remote_exists = list(remote.cache_exists(md5s))  
                                                             
        progress.finish_target(title)                        
                                                             
        self._fill_statuses(ret, local_exists, remote_exists)
                                                             
        self._log_missing_caches(ret)                        
                                                             
        return ret                                           
```

Reply:
neat, @efiop :+1: